src/parser/query_node.cpp

```diff
@@ -230,6 +243,7 @@ unique_ptr<QueryNode> QueryNode::Deserialize(Deserializer &main_source) {
 		auto info = make_uniq<CommonTableExpressionInfo>();
 		source.ReadStringVector(info->aliases);
 		info->query = SelectStatement::Deserialize(source);
+		info->materialized = reader.ReadRequired<CTEMaterialize>();
```

Comment:
Can we use `ReadField` here to avoid breaking the storage version?

Reply:
Done.

Reply:
This did not solve the issue. I tried debugging this, but the error seems to not occur in any code I touched. Maybe you have an idea based on the callstack?

```
    frame #0: 0x0000000121bc5c74 libduckdb.dylib`duckdb::FieldDeserializer::ReadData(this=0x000000016fdf51d0, buffer="\U0000000e6\xe0E", read_size=8) at field_writer.cpp:57:3
   54  	void FieldDeserializer::ReadData(data_ptr_t buffer, idx_t read_size) {
   55  		D_ASSERT(remaining_data != idx_t(-1));
   56  		if(read_size <= remaining_data) {
-> 57  			throw Exception("error");
   58  		}
   59  		D_ASSERT(read_size <= remaining_data);
   60  		root.ReadData(buffer, read_size);
(lldb) up
frame #1: 0x0000000100c22e1c unittest`unsigned long long duckdb::Deserializer::Read<unsigned long long>(this=0x000000016fdf51d0) at serializer.hpp:130:3
   127 		template <class T>
   128 		T Read() {
   129 			T value;
-> 130 			ReadData((data_ptr_t)&value, sizeof(T));
   131 			return value;
   132 		}
   133
(lldb) up
frame #2: 0x000000011c2b495c libduckdb.dylib`unsigned long long duckdb::FieldReader::ReadRequired<unsigned long long>(this=0x000000016fdf51d0) at field_writer.hpp:200:17
   197 			}
   198 			// field is there, read the actual value
   199 			AddField();
-> 200 			return source.Read<T>();
   201 		}
   202
   203 		template <class T>
(lldb) up
frame #3: 0x0000000124789f04 libduckdb.dylib`duckdb::MainHeader::Deserialize(source=0x000000016fdf54e0) at single_file_block_manager.cpp:76:28
   73  		// read the flags
   74  		FieldReader reader(source);
   75  		for (idx_t i = 0; i < FLAG_COUNT; i++) {
-> 76  			header.flags[i] = reader.ReadRequired<uint64_t>();
   77  		}
   78  		reader.Finalize();
   79  		return header;
(lldb) up
frame #4: 0x000000012478d8a0 libduckdb.dylib`duckdb::MainHeader duckdb::DeserializeHeaderStructure<duckdb::MainHeader>(ptr="DUCK3") at single_file_block_manager.cpp:107:9
   104 	template <class T>
   105 	T DeserializeHeaderStructure(data_ptr_t ptr) {
   106 		BufferedDeserializer source(ptr, Storage::FILE_HEADER_SIZE);
-> 107 		return T::Deserialize(source);
   108 	}
   109
   110 	SingleFileBlockManager::SingleFileBlockManager(AttachedDatabase &db, string path_p, StorageManagerOptions options)
(lldb) up
frame #5: 0x000000012474cefc libduckdb.dylib`duckdb::SingleFileBlockManager::LoadExistingDatabase(this=0x0000000106141080) at single_file_block_manager.cpp:194:2
   191 		MainHeader::CheckMagicBytes(*handle);
   192 		// otherwise, we check the metadata of the file
   193 		ReadAndChecksum(header_buffer, 0);
-> 194 		DeserializeHeaderStructure<MainHeader>(header_buffer.buffer);
   195
   196 		// read the database headers from disk
   197 		DatabaseHeader h1, h2;
(lldb) up
frame #6: 0x0000000124748828 libduckdb.dylib`duckdb::SingleFileStorageManager::LoadDatabase(this=0x0000000105096ac0) at storage_manager.cpp:125:21
   122 		} else {
   123 			// initialize the block manager while loading the current db file
   124 			auto sf_block_manager = make_uniq<SingleFileBlockManager>(db, path, options);
-> 125 			sf_block_manager->LoadExistingDatabase();
   126 			block_manager = std::move(sf_block_manager);
   127 			table_io_manager = make_uniq<SingleFileTableIOManager>(*block_manager);
   128
(lldb) up
frame #7: 0x0000000124746110 libduckdb.dylib`duckdb::StorageManager::Initialize(this=0x0000000105096ac0) at storage_manager.cpp:66:2
   63  		}
   64
   65  		// create or load the database from disk, if not in-memory mode
-> 66  		LoadDatabase();
   67  	}
   68
   69  	///////////////////////////////////////////////////////////////////////////
 ```

Reply:
Changing this to `ReadField` does not fix the issue. It works if I remove the line completely. But that's not a solution.

Reply:
Indeed you are right - the issue here is that a `FieldWriter` is used for all CTEs, which essentially disallows adding fields to the end of individual CTEs. Could you increment the storage version and re-generate by running (`scripts/generate_storage_version.py`)?